plugins/modules/aci_system_connectivity_prefs.py

```diff
@@ -0,0 +1,250 @@
+  returned: failure
+  type: dict
+  sample:
+    {
+      "code": "120",
+      "text": "unknown property value foo, name interfacePref, class mgmtConnectivityPrefs [(Dn0)] Dn0=, "
+    }
+raw:
+  description: The raw output returned by the APIC REST API (xml or json)
+  returned: parse error
+  type: str
+  sample: 
+    {
+      "totalCount": "1",
+      "imdata": [
+          {
+              "error": {
+                  "attributes": {
+                      "code": "120",
+                      "text": "unknown property value foo, name interfacePref, class mgmtConnectivityPrefs [(Dn0)] Dn0=, "
+                  }
+              }
+          }
+      ]
+    }
+sent:
+  description: The actual/minimal configuration pushed to the APIC
+  returned: info
+  type: list
+  sample:
+    {
+        "mgmtConnectivityPrefs": { 
+            "attributes": {
+                "interfacePref": "inband"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "mgmtConnectivityPrefs": {
+                "attributes": {
+                    "annotation": "",
+                    "descr": "",
+                    "dn": "uni/fabric/connectivityPrefs",
+                    "interfacePref": "ooband",
+                    "name": "default",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "mgmtConnectivityPrefs": { 
+            "attributes": {
+                "interfacePref": "inband"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        interface_pref=dict(type="str"),
```

Comment:
I think we should validate the input before sending it to APIC. Also depending on aliases chosen we should add here.

```suggestion
        interface_preference=dict(type="str", choices=["inband", "ooband"] aliases=["interface_pref", "int_pref", "external_connection"]),
```